libvast/vast/coder.hpp

```diff
@@ -588,11 +680,69 @@ class multi_level_coder
+    static_assert(sizeof(value_type) <= sizeof(uint64_t),
+                  "value_type too large to be normalized to 64 bit");
+    auto base_normalized = std::vector<uint64_t>{};
+    base_normalized.reserve(value.base_.size());
+    for (const auto& base_value : value.base_)
+      base_normalized.emplace_back(static_cast<uint64_t>(base_value));
+    const auto base_offset
+      = fbs::coder::detail::CreateBaseDirect(builder, &base_normalized);
+    const auto xs_offset = builder.CreateVector(
+      reinterpret_cast<const uint64_t*>(value.xs_.data()), value.xs_.size());
+    constexpr auto coder_type
+      = is_singleton_coder<Coder>::value ? fbs::coder::Coder::singleton
+        : std::disjunction_v<is_equality_coder<Coder>, is_range_coder<Coder>,
+                             is_bitslice_coder<Coder>>
+          ? fbs::coder::Coder::vector
+          : fbs::coder::Coder::multi_level;
+    auto coder_offsets = std::vector<flatbuffers::Offset<fbs::Coder>>{};
+    coder_offsets.reserve(value.coders_.size());
+    for (const auto& coder : value.coders_)
+      coder_offsets.emplace_back(
+        fbs::CreateCoder(builder, coder_type, pack(builder, coder).Union()));
+    const auto coders_offset = builder.CreateVector(coder_offsets);
+    return fbs::coder::CreateMultiLevelCoder(builder, base_offset, xs_offset,
+                                             coders_offset);
+  }
+
+  friend caf::error
+  unpack(const fbs::coder::MultiLevelCoder& from, multi_level_coder& to) {
+    auto base_values = std::vector<value_type>{};
+    base_values.reserve(from.base()->values()->size());
+    for (const auto& base_value : *from.base()->values())
+      base_values.emplace_back(static_cast<value_type>(base_value));
+    to.base_ = vast::base{std::move(base_values)};
+    to.xs_.reserve(from.xs()->size());
+    to.xs_.insert(to.xs_.end(), from.xs()->begin(), from.xs()->end());
+    using concrete_coder_type = std::conditional_t<
+      is_singleton_coder<Coder>::value, fbs::coder::SingletonCoder,
+      std::conditional_t<
+        std::disjunction_v<is_equality_coder<Coder>, is_range_coder<Coder>,
+                           is_bitslice_coder<Coder>>,
+        fbs::coder::VectorCoder, fbs::coder::MultiLevelCoder>>;
+    to.coders_.reserve(from.coders()->size());
+    for (const auto& from_coder : *from.coders()) {
+      const auto* from_concrete_coder
+        = from_coder->coder_as<concrete_coder_type>();
+      if (!from_concrete_coder)
+        return caf::make_error(
+          ec::logic_error, "invalid vast.fbs.coder.MultiLevelCoder coder type");
+      auto to_coder = Coder{};
+      if (auto err = unpack(*from_concrete_coder, to_coder))
+        return err;
+      to.coders_.emplace_back(std::move(to_coder));
+    }
+    return caf::none;
+  }
+
 private:
   void init() {
     VAST_ASSERT(base_.well_defined());
-    xs_.resize(base_.size()),
-    coders_.resize(base_.size());
+    xs_.resize(base_.size()), coders_.resize(base_.size());
```

Comment:
This doesn't make sense to me. Why separate the statements by comma?

Reply:
They already were separated by comma from a 4 years old commit, and was just reformatted by clang-format here. I'll `s/,/;`.